Add conda development install guidance (#660)
pyOpenSci/python-package-guide: maintain-automate/dev-installs.md

So... why pip? The short answer is that conda and pip are doing different jobs here.

The slightly longer, mostly apologetic, answer is this is just sort of the current ergonomics of how python packaging works and, honestly? Most of us have turned this confusing pain point into muscle memory. But not you. You're new here. And you're like... wat? And you're totally justified to feel this way.

So, what is happening here? `conda` manages the environment: the Python runtime, compiled libraries, command line tools, and the packages your project depends on. This is stuff that you've already been doing and you're comfortable with (or at least familiar with). `pip` is doing one Python-packaging-specific job: installing your local package into the active environment.

In editable mode, the `-e` flag, `pip` connects the active environment to the source files you are editing. And... why exactly is that useful?

> It's useful because it gives you pretty quick and simple development loop:

Reviewer: ```suggestion
It's useful because it gives you a pretty quick development loop:
```

1. Edit your code in your editor.
2. Run it from a terminal, test suite, or Jupyter notebook.
3. Edit the code again.
4. Run it again without reinstalling your package.

So the goal is not to switch from conda to pip. The goal is to keep using your conda environment while making your local package importable inside that environment.

## Should I use pip for everything now?

Probably not?

If conda is already working well for your project, keep using conda to manage the environment. Use pip only for this one task: installing your local package in editable mode.

If you are curious about other tools like uv, pixi, Hatch, or pip-only workflows, see [Environment Managers](environment-managers.md). Those tools can be great choices. But you do not need to switch tools just to develop your local package inside a conda environment.

> As a final note, people in the conda ecosystem are actively working on better conda/pip interop. In the future, this workflow may become less awkward. For now, `python -m pip install -e . --no-deps` is the standard bridge.

Reviewer: ```suggestion
As a final note, people in the conda ecosystem are actively working on better conda/pip interoperability. In the future, this workflow may become less awkward. For now, `python -m pip install -e . --no-deps` is the standard bridge.
```
